### PR TITLE
Feat/exp 3

### PR DESCRIPTION
**Pull request by eveleighoj**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Alter and improve the expectations work. It has become clear that the inputs and outputs were too complicated to take in and use. this introduces a more simplified out put and input based on configuration files.

## Related Tickets & Documents

- Ticket Link
- Related Issue #
- Closes #

## QA Instructions, Screenshots, Recordings

Tests should replicate al the relevant behaviour so worth reviewing them to ensure no other cases are needed. 

this will be supported by changes to makerules, will update with a PR link

this will be supported by a branch of collection task which can be ran locally to check. This won't need a PR but will set up relevant environment variables to easily run.

_Please replace this line with instructions on how to test your changes, a note
on the devices and browsers this has been tested on, as well as any relevant
images for UI changes._

## Added/updated tests?
_We encourage you to keep the code coverage percentage at 80% and above._

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [optional] Are there any dependencies on other PRs or Work?

Will need a demonstration to our team, DM team and finally to Providers

### CODE REVIEW

File: digital_land/cli.py [hunk 1]
```diff
@@ -287,16 +286,57 @@ def collection_add_source_cmd(ctx, collection, endpoint_url, collection_dir):
...
+@click.option(
+    "--organisation-path",
+    type=click.Path(),
+    help="path to the organisation data for the organisation class",
+    required=True,
+)
+@click.option(
+    "--specification-dir",
+    type=click.Path(),
+    help="directory  containing the specification",
```
Comment (cjohns-scottlogic) [inline digital_land/cli.py:322]: (nitpick) there are two spaces after the word directory in the help text

File: digital_land/cli.py [hunk 2]
```diff
@@ -287,16 +286,57 @@ def collection_add_source_cmd(ctx, collection, endpoint_url, collection_dir):
...
+)
+def expectations_run_dataset_checkpoint(
+    dataset,
+    file_path,
+    log_dir,
+    configuration_path,
+    organisation_path,
+    specification_dir,
+):
     from digital_land.expectations.commands import run_dataset_checkpoint
```
Comment (cjohns-scottlogic) [inline digital_land/cli.py:333]: Does this need to be imported here, rather than the top of the file?

File: digital_land/expectations/issue.py [hunk 3]
```diff
@@ -1,6 +1,7 @@
 """
 Module containing the dataclasses representing the different
 expectation errors that can be recorded when running expectations
+TOBEREMOVED
```
Comment (cjohns-scottlogic) [inline digital_land/expectations/issue.py:4]: Does this need to be removed? :)

File: digital_land/expectations/log.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
+import csv
+import json
+import pyarrow as pa
+import pyarrow.parquet as pq
+
+from pathlib import Path
+
+
+# TODO Review and connect to centralised Log classes do we need our own class for this
+class ExpectationLog:
```
Comment (cjohns-scottlogic) [inline digital_land/expectations/log.py:10]: Can this inherit from the base Log class? (digital_land/log.py)

File: digital_land/expectations/operation.py [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+
+    # compare expected to actual
+    # Define comparison rules
+    comparison_rules = {
+        "equals_to": actual == expected,
+        "greater_than": actual > expected,
+        "greater_than_or_equal_to": actual >= expected,
+        "less_than": actual < expected,
+        "less_than_or_equal_to": actual <= expected,
+    }
```
Comment (cjohns-scottlogic) [inline digital_land/expectations/operation.py:96]: do we need a not_equal_to ?

File: digital_land/package/sqlite.py [hunk 6]
```diff
@@ -59,6 +59,7 @@ def disconnect(self):
         self.connection.close()
 
     def create_table(self, table, fields, key_field=None, unique=None):
+        logging.debug(fields)
```
Comment (cjohns-scottlogic) [inline digital_land/package/sqlite.py:62]: is this stray debug?

File: digital_land/cli.py (hunk 1 above)
Comment (eveleighoj) [reply digital_land/cli.py:322]: Updated

File: digital_land/cli.py (hunk 2 above)
Comment (eveleighoj) [reply digital_land/cli.py:333]: I think it was here to stop spatialite being imported but that doesn't matter anymore

File: digital_land/expectations/issue.py (hunk 3 above)
Comment (eveleighoj) [reply digital_land/expectations/issue.py:4]: Will delete

File: digital_land/expectations/log.py (hunk 4 above)
Comment (eveleighoj) [reply digital_land/expectations/log.py:10]: I thikn yes in the. future but let's leave as separate for now and combine at a later date

File: digital_land/expectations/operation.py (hunk 5 above)
Comment (eveleighoj) [reply digital_land/expectations/operation.py:94]: Added

File: digital_land/package/sqlite.py (hunk 6 above)
Comment (eveleighoj) [reply digital_land/package/sqlite.py:62]: Removed

**Review by cjohns-scottlogic — APPROVED** [review]

**Pull request merged**